src/main/scala/com/databricks/spark/sql/perf/results.scala

```diff
@@ -97,3 +102,48 @@ case class BreakdownResult(
     delta: Double)
 
 case class Failure(className: String, message: String)
+
+case class MLTestParameters(
+    numFeatures: Option[Int] = None,
+    numExamples: Option[Long] = None,
+    numTestExamples: Option[Long] = None,
+    numPartitions: Option[Int] = None,
+    randomSeed: Option[Int] = None)
+
+object MLTestParameters {
+
+  val empty = MLTestParameters()
+}
+
+// KEEP ARGUMENTS SORTED BY NAME
+case class ExtraMLTestParameters(
+    ldaDocLength: Option[Int] = None,
+    ldaNumVocabulary: Option[Int] = None,
+    ldaNumTopics: Option[Int] = None,
+    ldaOptimizer: Option[String] = None,
```

Comment:
For common params, I'd just share 1 param: "optimizer" instead of "ldaOptimizer"